### PR TITLE
Support piping contents of a file via stdin for checking

### PR DESCRIPTION
**Pull request by aclemons**
This will allow linting via ALE in vim. I have a pull request ready for that once this is merged. Other editors can also use this feature to implement linting of erubies via ruumba.

I've also done some refactoring and added spec coverage for the things I promised quite a while ago in https://github.com/ericqweinstein/ruumba/pull/11#issuecomment-353236532.

### CODE REVIEW

**Comment by aclemons** [comment]
Any updates on this one @ericqweinstein ?

**Comment by ericqweinstein** [comment]
@aclemons Thanks so much for your patience—my son was born recently and I'm starting a company, so things have been super hectic on my end. I'll try to review this week 👍

**Comment by ericqweinstein** [comment]
@aclemons Thanks again for your patience—if you can resolve conflicts, this LGTM.

**Comment by aclemons** [comment]
@ericqweinstein I've rebase and tidied up the conflicts now. Should be ready to go again.

File: .ruby-version [hunk 1]
```diff
@@ -1 +1 @@
-2.4.1
+2.3.7
```
Comment (ericqweinstein) [inline .ruby-version]: @aclemons Remind me, why the downgrade to 2.3.7 again? (If anything, we should try to move ahead to 2.5.2 or 2.5.3 if we can)

File: lib/ruumba/parser.rb [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+  class Parser
+    # The regular expression to capture interpolated Ruby.
+    ERB_REGEX = /<%[-=]?(.*?)-?%>/m
+
+    # Extracts Ruby code from an ERB template.
+    # @return [String] The extracted ruby code
+    def extract(contents)
+      file_text, matches = parse(contents)
+
+      extracted_ruby = +''
```
Comment (ericqweinstein) [inline lib/ruumba/parser.rb]: Why the `+` here?

File: spec/ruumba/iterators_spec.rb [hunk 3]
```diff
@@ -0,0 +1,63 @@
+# @author Eric Weinstein <eric.q.weinstein@gmail.com>
```
Comment (ericqweinstein) [inline spec/ruumba/iterators_spec.rb]: @aclemons You can include yourself as the author here if you like

File: .ruby-version (hunk 1 above)
Comment (aclemons) [reply .ruby-version]: I was just targeting the oldest currently supported ruby version.

File: lib/ruumba/parser.rb (hunk 2 above)
Comment (aclemons) [reply lib/ruumba/parser.rb]: I had added `# frozen_string_literal: true` to the file, so I needed to use `+''` to initialise an unfrozen string.

File: spec/ruumba/iterators_spec.rb (hunk 3 above)
Comment (aclemons) [reply spec/ruumba/iterators_spec.rb]: Ok, added now (force pushed).

File: .ruby-version (hunk 1 above)
Comment (ericqweinstein) [reply .ruby-version]: Got it. I'm okay with this for the time being. 👍

**Pull request merged**